test/extended/single_node/topology.go

```diff
@@ -67,61 +93,88 @@ func isInfrastructureDeployment(deployment appsv1.Deployment) bool {
 	return false
 }
 
-func validateReplicas(deployment appsv1.Deployment,
-	controlPlaneTopology, infraTopology v1.TopologyMode, failureAllowed bool) {
-	if isInfrastructureDeployment(deployment) {
+func validateReplicas(name, namespace string, replicas int, failureAllowed bool) {
+	if !failureAllowed {
+		Expect(replicas).To(Equal(1),
+			"%s in %s namespace expected to have 1 replica but got %d", name, namespace, replicas)
+	} else {
+		if replicas == 1 {
+			t := GinkgoT()
+			t.Logf("%s in namespace %s has one replica, consider taking it off the topology allow-list",
```

Comment:
I'd suggest using `framework.Logf` instead of this construct to be consistent with other tests.